htlcswitch/link.go

```diff
@@ -3877,14 +3879,14 @@ func (l *channelLink) processExitHop(pd *lnwallet.PaymentDescriptor,
 
 // settleHTLC settles the HTLC on the channel.
 func (l *channelLink) settleHTLC(preimage lntypes.Preimage,
-	pd *lnwallet.PaymentDescriptor) error {
+	htlcIndex uint64, sourceRef channeldb.AddRef) error {
```

Comment:
Trying to eliminate bad pointer usage. I'd rather propagate it the other way.

Reply:
Yeah upon second inspection here I can't propagate value usage naively here because the call stack for this value misuses pointers to simulate options. I could fix it all in one commit but I think that gets further afield in this PR than I'd like. I actually prefer the tension here as it will motivate future refactors to resolve it in their own focused PRs.